moonraker/plugins/history.py

```diff
@@ -0,0 +1,253 @@
+# History cache for printer jobs
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+import json, logging, time
+from tornado.ioloop import IOLoop
+
+SAVE_INTERVAL = 5
+
+HIST_NAMESPACE = "history"
+METADATA_KEYS = ['estimated_time','modified','slicer','slicer_version']
+
+class History:
+    def __init__(self, config):
+        self.server = config.get_server()
+        self.database = self.server.lookup_plugin("database")
+        self.gcdb = self.database.wrap_namespace("gcode_metadata",
+            parse_keys=False)
+        self.current_job = None
+        self.file_metadata = None
+        self.jobs = {}
+        self.job_id = -1
+        self.last_update_time = 0
+        self.print_stats = {}
+
+        self.server.register_event_handler(
+            "server:klippy_ready", self._init_ready)
+        self.server.register_event_handler(
+            "server:status_update", self._status_update)
+
+
+        self.server.register_endpoint(
+            "/printer/history/list", ['GET'], self._handle_jobs_list)
+        self.server.register_endpoint(
+            "/printer/history/delete", ['DELETE'], self._handle_job_delete)
```

Comment:
I think these endpoints should be in the `/server` path.  The `/printer` path is reserved for endpoints registered by Klippy, the exception to this are the aliases registered in `klippy_apis`.  This prevents a situation where API collisions can occur.